src/Pulp/Files.purs

```diff
@@ -54,9 +63,20 @@ testGlobs :: Options -> AffN (Set String)
 testGlobs = globsFromOption "testPath"
 
 dependencyGlobs :: Options -> AffN (Set String)
-dependencyGlobs =
-  globsFromOption' (\path -> Path.concat [path, "purescript-*", "src"])
-                   "dependencyPath"
+dependencyGlobs opts = do
+  Project pro <- getOption' "_project" opts
+  -- If project file has a `set` property we assume it's a psc-package project file
+  case runExcept (readProp "set" pro.projectFile >>= readString) of
+    Right _ -> pscPackageGlobs
+    _ -> globsFromOption' (\path -> Path.concat [path, "purescript-*", "src"]) "dependencyPath" opts
+
+pscPackageGlobs :: AffN (Set String)
+pscPackageGlobs = do
+  execQuiet "psc-package" ["sources"] Nothing <#>
+  Set.fromFoldable <<<
+  -- Strip the glob.purs suffixes just to append them later so it plays well with the other globs
+  mapMaybe (stripSuffix (Pattern "/**/*.purs")) <<<
+  split (unsafeRegex "\r\n|\n" noFlags)
```

Comment:
I found this a little difficult to read, since I am not sure which has higher precendence out of `<#>` and `<<<`. Do you think it might make sense to split off a part of this expression and give it a name in a `where` binding or something like that?

Also, I don't think regex should be necessary for this and in particular I'd like to avoid use of `unsafeRegex`. Would something like `String.split (Pattern "\r\n") >=> String.split (Pattern "\n")` work instead?